gitclone/checkout.go

```diff
@@ -171,20 +167,10 @@ func createCheckoutStrategy(checkoutMethod CheckoutMethod, cfg Config, patch pat
 				patchFile: patchFile,
 			}, nil
 		}
-	case CheckoutForkPRManualMergeMethod:
-		{
-			params, err := NewForkPRManualMergeParams(cfg.Branch, cfg.PRRepositoryURL, cfg.BranchDest)
-			if err != nil {
-				return nil, err
-			}
-
-			return checkoutForkPRManualMerge{
-				params: *params,
-			}, nil
-		}
 	case CheckoutPRManualMergeMethod:
 		{
-			params, err := NewPRManualMergeParams(cfg.Branch, cfg.Commit, cfg.BranchDest)
+			isFork := isFork(cfg.RepositoryURL, cfg.PRRepositoryURL)
+			params, err := NewPRManualMergeParams(isFork, cfg.Branch, cfg.Commit, cfg.PRRepositoryURL, cfg.BranchDest)
```

Comment:
if we only set `PRRepositoryURL` in case of a fork, we could avoid the `isFork` input.

Reply:
I was not sure about only either one or the other exists. For instance, `PR - no fork - manual merge: repo is the same with different schem` test sets both the `PRMergeBranch` and the `Commit` parameters. Since the `RepositoryURL` is not passed to the param instantiation function, I thought the safest way is just to pass this info as an `isFork` parameter.
However, if we could verify that only one of them is set, and therefore we can use it to identify a fork, I'm happy to remove and simplify it.

Reply:
I ment somthing like this:

```
			prRepositoryURL := ""
			isFork := isFork(cfg.RepositoryURL, cfg.PRRepositoryURL)
			if isFork {
				prRepositoryURL = cfg.PRRepositoryURL
			}
			params, err := NewPRManualMergeParams(cfg.Branch, cfg.Commit, prRepositoryURL, cfg.BranchDest)
			if err != nil {
				return nil, err
			}
```

This way the presents of prRepositoryURL indicates if the PR head repo is a fork.

In the case of the `PR - no fork - manual merge: repo is the same with different scheme` test `prRepositoryURL` would remain blank.

Reply:
Sounds good!